File: Python/최소직사각형.py
def solution(sizes):
    data = list(map(lambda x: x if x[0] > x[1] else [x[1],x[0]],sizes))
    return max(data, key=lambda x:x[0])[0] * max(data, key=lambda x:x[1])[1]


solution = lambda sizes: max(sum(sizes, [])) * max(min(size) for size in sizes)

File: Kotlin/최소직사각형.kt
fun solution(sizes: Array<IntArray>): Int {
        var maxW = 0; var maxH = 0;
        sizes.forEach {i ->
            if (i[0] < i[1]) i[0] = i[0] xor i[1] xor i[0].also { i[1] = it }
            maxW = if(i[0] > maxW) i[0] else maxW
            maxH = if(i[1] > maxH) i[1] else maxH
        }
        return maxW * maxH
    }